glean-core/uniffi/src/core/mod.rs

```diff
@@ -222,4 +225,71 @@ impl Glean {
         let metric = ExperimentMetric::new(self, experiment_id);
         metric.test_get_value(self)
     }
+
+    /// Sets a debug view tag.
+    ///
+    /// This will return `false` in case `value` is not a valid tag.
+    ///
+    /// When the debug view tag is set, pings are sent with a `X-Debug-ID` header with the value of the tag
+    /// and are sent to the ["Ping Debug Viewer"](https://mozilla.github.io/glean/book/dev/core/internal/debug-pings.html).
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - A valid HTTP header value. Must match the regex: "[a-zA-Z0-9-]{1,20}".
+    pub fn set_debug_view_tag(&mut self, value: &str) -> bool {
+        self.debug.debug_view_tag.set(value.into())
+    }
+
+    /// Return the value for the debug view tag or [`None`] if it hasn't been set.
+    ///
+    /// The `debug_view_tag` may be set from an environment variable
+    /// (`GLEAN_DEBUG_VIEW_TAG`) or through the [`set_debug_view_tag`] function.
+    pub(crate) fn debug_view_tag(&self) -> Option<&String> {
+        self.debug.debug_view_tag.get()
+    }
```

Comment:
I assume the need for the "property accessor" pattern here is to control access and only expose the `set` to UniFFI?

Reply:
It's only used by the (not-yet-included) ping maker: https://github.com/mozilla/glean/blob/uniffi-settags-api/glean-core/src/ping/mod.rs#L194-L200